### PR TITLE
Properly register UTXOs from migration funds

### PR DESCRIPTION
**Pull request by diega**
I suggest to first merge #329 so the integration tests became runnable again

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java [hunk 1]
```diff
@@ -628,10 +630,22 @@ private void processSigning(long executionBlockNumber, BtcECKey federatorPublicK
...
-                txIn.setScriptSig(inputScript);
-                logger.debug("Tx input for tx {} signed.", new Sha3Hash(rskTxHash));
+                try {
+                    int sigIndex = inputScript.getSigInsertionIndex(sighash, federatorPublicKey);
+                    inputScript = ScriptBuilder.updateScriptWithSignature(inputScript, txSig.encodeToBitcoin(), sigIndex, 1, 1);
+                    txIn.setScriptSig(inputScript);
+                    logger.debug("Tx input for tx {} signed.", new Sha3Hash(rskTxHash));
+                } catch (IllegalStateException e) {
+                    Federation retiringFederation = getRetiringFederation();
+                    if (getActiveFederation().getPublicKeys().contains(federatorPublicKey)) {
```
Comment (amendelzon) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java]: According to what we've been looking at with @tinchou, you should probably be using the safer `Federation::hasPublicKey` instead. Same below with the retiring federation.

File: rskj-core/src/main/java/co/rsk/rpc/modules/personal/PersonalModuleWalletEnabled.java [hunk 2]
```diff
@@ -111,7 +111,9 @@ public String importRawKey(String key, String passphrase) {
             byte[] address = this.wallet.addAccountWithPrivateKey(Hex.decode(key), passphrase);
             return s = TypeConverter.toJsonHex(address);
         } finally {
-            LOGGER.debug("personal_importRawKey(*****): {}", s);
+            char[] hiddenKey = new char[key.length()];
+            Arrays.fill(hiddenKey, '*');
+            LOGGER.debug("personal_importRawKey({}, {}): {}", new String(hiddenKey), passphrase, s);
```
Comment (amendelzon) [inline rskj-core/src/main/java/co/rsk/rpc/modules/personal/PersonalModuleWalletEnabled.java]: We should probably be obfuscating the `passphrase` as well.

File: rskj-core/src/main/java/co/rsk/rpc/modules/personal/PersonalModuleWalletEnabled.java [hunk 3]
```diff
@@ -121,7 +123,7 @@ public String sendTransaction(Web3.CallArguments args, String passphrase) throws
         try {
             return s = sendTransaction(args, getAccount(args.from, passphrase));
         } finally {
-            LOGGER.debug("eth_sendTransaction(" + args + "): " + s);
+            LOGGER.debug("eth_sendTransaction({}, {}): {}", args, passphrase, s);
```
Comment (amendelzon) [inline rskj-core/src/main/java/co/rsk/rpc/modules/personal/PersonalModuleWalletEnabled.java]: Same here.

**Comment by rskops** [comment]
SonarQube analysis reported 168 issues
* ![MAJOR][MAJOR] 109 major
* ![MINOR][MINOR] 35 minor
* ![INFO][INFO] 24 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [BridgeRegTestConstants.java#L41](https://github.com/rsksmart/rskj/blob/ad3be6f07af735103cb473a30dc5fe3a24fbb821/rskj-core/src/main/java/co/rsk/config/BridgeRegTestConstants.java#L41): Remove this unused "logger" private field. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [Bridge.java#L1](https://github.com/rsksmart/rskj/blob/ad3be6f07af735103cb473a30dc5fe3a24fbb821/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L1): Class co.rsk.peg.Bridge defines static field "co.rsk.peg.Bridge.functionCostMap" which appears to allow memory bloat [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APMB_POSSIBLE_MEMORY_BLOAT)
1. ![MAJOR][MAJOR] [Bridge.java#L267](https://github.com/rsksmart/rskj/blob/ad3be6f07af735103cb473a30dc5fe3a24fbb821/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L267): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [Bridge.java#L298](https://github.com/rsksmart/rskj/blob/ad3be6f07af735103cb473a30dc5fe3a24fbb821/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L298): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Bridge.java#L308](https://github.com/rsksmart/rskj/blob/ad3be6f07af735103cb473a30dc5fe3a24fbb821/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L308): Extract this nested try block into a separate method. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1141)
1. ![MAJOR][MAJOR] [Bridge.java#L313](https://github.com/rsksmart/rskj/blob/ad3be6f07af735103cb473a30dc5fe3a24fbb821/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L313): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Bridge.java#L330](https://github.com/rsksmart/rskj/blob/ad3be6f07af735103cb473a30dc5fe3a24fbb821/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L330): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L334](https://github.com/rsksmart/rskj/blob/ad3be6f07af735103cb473a30dc5fe3a24fbb821/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L334): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L342](https://github.com/rsksmart/rskj/blob/ad3be6f07af735103cb473a30dc5fe3a24fbb821/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L342): Remove this unused method parameter "args". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [Bridge.java#L350](https://github.com/rsksmart/rskj/blob/ad3be6f07af735103cb473a30dc5fe3a24fbb821/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L350): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 168 issues
* ![MAJOR][MAJOR] 109 major
* ![MINOR][MINOR] 35 minor
* ![INFO][INFO] 24 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [BridgeRegTestConstants.java#L41](https://github.com/rsksmart/rskj/blob/c964dc3ce4f374a927c4912d3a46556c92c6aee8/rskj-core/src/main/java/co/rsk/config/BridgeRegTestConstants.java#L41): Remove this unused "logger" private field. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [Bridge.java#L1](https://github.com/rsksmart/rskj/blob/c964dc3ce4f374a927c4912d3a46556c92c6aee8/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L1): Class co.rsk.peg.Bridge defines static field "co.rsk.peg.Bridge.functionCostMap" which appears to allow memory bloat [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APMB_POSSIBLE_MEMORY_BLOAT)
1. ![MAJOR][MAJOR] [Bridge.java#L267](https://github.com/rsksmart/rskj/blob/c964dc3ce4f374a927c4912d3a46556c92c6aee8/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L267): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [Bridge.java#L298](https://github.com/rsksmart/rskj/blob/c964dc3ce4f374a927c4912d3a46556c92c6aee8/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L298): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Bridge.java#L308](https://github.com/rsksmart/rskj/blob/c964dc3ce4f374a927c4912d3a46556c92c6aee8/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L308): Extract this nested try block into a separate method. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1141)
1. ![MAJOR][MAJOR] [Bridge.java#L313](https://github.com/rsksmart/rskj/blob/c964dc3ce4f374a927c4912d3a46556c92c6aee8/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L313): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Bridge.java#L330](https://github.com/rsksmart/rskj/blob/c964dc3ce4f374a927c4912d3a46556c92c6aee8/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L330): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L334](https://github.com/rsksmart/rskj/blob/c964dc3ce4f374a927c4912d3a46556c92c6aee8/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L334): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L342](https://github.com/rsksmart/rskj/blob/c964dc3ce4f374a927c4912d3a46556c92c6aee8/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L342): Remove this unused method parameter "args". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [Bridge.java#L350](https://github.com/rsksmart/rskj/blob/c964dc3ce4f374a927c4912d3a46556c92c6aee8/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L350): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 168 issues
* ![MAJOR][MAJOR] 109 major
* ![MINOR][MINOR] 35 minor
* ![INFO][INFO] 24 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [BridgeRegTestConstants.java#L41](https://github.com/rsksmart/rskj/blob/78e3f888bb9c7831196e9a2f6ba508b5e4c27499/rskj-core/src/main/java/co/rsk/config/BridgeRegTestConstants.java#L41): Remove this unused "logger" private field. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [Bridge.java#L1](https://github.com/rsksmart/rskj/blob/78e3f888bb9c7831196e9a2f6ba508b5e4c27499/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L1): Class co.rsk.peg.Bridge defines static field "co.rsk.peg.Bridge.functionCostMap" which appears to allow memory bloat [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APMB_POSSIBLE_MEMORY_BLOAT)
1. ![MAJOR][MAJOR] [Bridge.java#L267](https://github.com/rsksmart/rskj/blob/78e3f888bb9c7831196e9a2f6ba508b5e4c27499/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L267): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [Bridge.java#L298](https://github.com/rsksmart/rskj/blob/78e3f888bb9c7831196e9a2f6ba508b5e4c27499/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L298): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Bridge.java#L308](https://github.com/rsksmart/rskj/blob/78e3f888bb9c7831196e9a2f6ba508b5e4c27499/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L308): Extract this nested try block into a separate method. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1141)
1. ![MAJOR][MAJOR] [Bridge.java#L313](https://github.com/rsksmart/rskj/blob/78e3f888bb9c7831196e9a2f6ba508b5e4c27499/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L313): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Bridge.java#L330](https://github.com/rsksmart/rskj/blob/78e3f888bb9c7831196e9a2f6ba508b5e4c27499/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L330): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L334](https://github.com/rsksmart/rskj/blob/78e3f888bb9c7831196e9a2f6ba508b5e4c27499/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L334): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L342](https://github.com/rsksmart/rskj/blob/78e3f888bb9c7831196e9a2f6ba508b5e4c27499/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L342): Remove this unused method parameter "args". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [Bridge.java#L350](https://github.com/rsksmart/rskj/blob/78e3f888bb9c7831196e9a2f6ba508b5e4c27499/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L350): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by diega** [comment]
✅ Add unit tests
✅ Obfuscate passphrase
✅ Use Federation::hasPublicKey

**Comment by rskops** [comment]
SonarQube analysis reported 168 issues
* ![MAJOR][MAJOR] 109 major
* ![MINOR][MINOR] 35 minor
* ![INFO][INFO] 24 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [BridgeRegTestConstants.java#L41](https://github.com/rsksmart/rskj/blob/0846cff8d2508d513e90bceffd8fbd83a2badd50/rskj-core/src/main/java/co/rsk/config/BridgeRegTestConstants.java#L41): Remove this unused "logger" private field. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [Bridge.java#L1](https://github.com/rsksmart/rskj/blob/0846cff8d2508d513e90bceffd8fbd83a2badd50/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L1): Class co.rsk.peg.Bridge defines static field "co.rsk.peg.Bridge.functionCostMap" which appears to allow memory bloat [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APMB_POSSIBLE_MEMORY_BLOAT)
1. ![MAJOR][MAJOR] [Bridge.java#L267](https://github.com/rsksmart/rskj/blob/0846cff8d2508d513e90bceffd8fbd83a2badd50/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L267): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [Bridge.java#L298](https://github.com/rsksmart/rskj/blob/0846cff8d2508d513e90bceffd8fbd83a2badd50/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L298): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Bridge.java#L308](https://github.com/rsksmart/rskj/blob/0846cff8d2508d513e90bceffd8fbd83a2badd50/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L308): Extract this nested try block into a separate method. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1141)
1. ![MAJOR][MAJOR] [Bridge.java#L313](https://github.com/rsksmart/rskj/blob/0846cff8d2508d513e90bceffd8fbd83a2badd50/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L313): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Bridge.java#L330](https://github.com/rsksmart/rskj/blob/0846cff8d2508d513e90bceffd8fbd83a2badd50/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L330): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L334](https://github.com/rsksmart/rskj/blob/0846cff8d2508d513e90bceffd8fbd83a2badd50/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L334): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L342](https://github.com/rsksmart/rskj/blob/0846cff8d2508d513e90bceffd8fbd83a2badd50/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L342): Remove this unused method parameter "args". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [Bridge.java#L350](https://github.com/rsksmart/rskj/blob/0846cff8d2508d513e90bceffd8fbd83a2badd50/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L350): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/co/rsk/rpc/modules/personal/PersonalModuleWalletEnabled.java [hunk 4]
```diff
@@ -196,4 +196,10 @@ private long convertFromJsonHexToLong(String x) throws Exception {
             throw new Exception("Incorrect hex syntax");
         return Long.parseLong(x.substring(2), 16);
     }
+
+    private static String obfuscateOutput(String sensitiveInformation) {
```
Comment (donequis) [inline rskj-core/src/main/java/co/rsk/rpc/modules/personal/PersonalModuleWalletEnabled.java]: What can all this prevent or be helpful for?

The only thing that I see is that whoever sees the log can read the length of the password.

File: rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java [hunk 5]
```diff
@@ -136,6 +136,16 @@ public static boolean isReleaseTx(BtcTransaction tx, Federation federation, Brid
         return false;
     }
 
+    public static boolean isMigrationTx(BtcTransaction btcTx, Federation activeFederation, Federation retiringFederation, Context btcContext, BridgeConstants bridgeConstants) {
+        if (retiringFederation == null) {
```
Comment (donequis) [inline rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java]: This is a false assumption. Do we have it reported at least?

File: rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java [hunk 6]
```diff
@@ -136,6 +136,16 @@ public static boolean isReleaseTx(BtcTransaction tx, Federation federation, Brid
...
+            return false;
+        }
+        boolean moveFromRetiring = isReleaseTx(btcTx, retiringFederation, bridgeConstants);
+        boolean moveToActive = isLockTx(btcTx, activeFederation, btcContext, bridgeConstants);
```
Comment (donequis) [inline rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java]: isLockTx has not a really cool name reaching this point

It is supposed to answer the question if someone is trying to move btc into rsk, but that's not exactly what it is doing. Also the method is overloaded in a slightly ugly way. We should have method for p2pkh, p2sh, etc. and check for fed inputs / outputs where it's needed. But maybe in a future pr.

I think this name is error prone.

File: rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java (hunk 5 above)
Comment (diega) [reply rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java]: Yes it's reported

File: rskj-core/src/main/java/co/rsk/rpc/modules/personal/PersonalModuleWalletEnabled.java (hunk 4 above)
Comment (diega) [reply rskj-core/src/main/java/co/rsk/rpc/modules/personal/PersonalModuleWalletEnabled.java]: I just removed this whole commit, it was just accessory

**Comment by rskops** [comment]
SonarQube analysis reported 148 issues
* ![MAJOR][MAJOR] 96 major
* ![MINOR][MINOR] 30 minor
* ![INFO][INFO] 22 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [BridgeRegTestConstants.java#L41](https://github.com/rsksmart/rskj/blob/50872ec10dcfe0ffe6a98a646ec04480135fea15/rskj-core/src/main/java/co/rsk/config/BridgeRegTestConstants.java#L41): Remove this unused "logger" private field. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [Bridge.java#L1](https://github.com/rsksmart/rskj/blob/50872ec10dcfe0ffe6a98a646ec04480135fea15/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L1): Class co.rsk.peg.Bridge defines static field "co.rsk.peg.Bridge.functionCostMap" which appears to allow memory bloat [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APMB_POSSIBLE_MEMORY_BLOAT)
1. ![MAJOR][MAJOR] [Bridge.java#L267](https://github.com/rsksmart/rskj/blob/50872ec10dcfe0ffe6a98a646ec04480135fea15/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L267): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [Bridge.java#L298](https://github.com/rsksmart/rskj/blob/50872ec10dcfe0ffe6a98a646ec04480135fea15/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L298): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Bridge.java#L308](https://github.com/rsksmart/rskj/blob/50872ec10dcfe0ffe6a98a646ec04480135fea15/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L308): Extract this nested try block into a separate method. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1141)
1. ![MAJOR][MAJOR] [Bridge.java#L313](https://github.com/rsksmart/rskj/blob/50872ec10dcfe0ffe6a98a646ec04480135fea15/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L313): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Bridge.java#L330](https://github.com/rsksmart/rskj/blob/50872ec10dcfe0ffe6a98a646ec04480135fea15/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L330): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L334](https://github.com/rsksmart/rskj/blob/50872ec10dcfe0ffe6a98a646ec04480135fea15/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L334): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L342](https://github.com/rsksmart/rskj/blob/50872ec10dcfe0ffe6a98a646ec04480135fea15/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L342): Remove this unused method parameter "args". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [Bridge.java#L350](https://github.com/rsksmart/rskj/blob/50872ec10dcfe0ffe6a98a646ec04480135fea15/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L350): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java [hunk 7]
```diff
@@ -330,6 +330,8 @@ public void registerBtcTransaction(Transaction rskTx, BtcTransaction btcTx, int
             // When is not guaranteed to be called in the chronological order, so a Federator can inform
             // b) In prod: Federator created a tx manually or the federation was compromised and some utxos were spent. Better not try to spend them.
             // Open problem: For performance removeUsedUTXOs() just removes 1 utxo
+        } else if (BridgeUtils.isMigrationTx(btcTx, getActiveFederation(), getRetiringFederation(), btcContext, bridgeConstants)) {
+            logger.debug("This is a migration tx {}", btcTx);
         } else {
             logger.warn("This is not a lock nor a release tx {}", btcTx);
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java]: Maybe update these logs? `".... nor a migration tx"`

**Comment by rskops** [comment]
SonarQube analysis reported 148 issues
* ![CRITICAL][CRITICAL] 20 critical
* ![MAJOR][MAJOR] 96 major
* ![MINOR][MINOR] 10 minor
* ![INFO][INFO] 22 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [Bridge.java#L297](https://github.com/rsksmart/rskj/blob/be1d47fc7508dac76d2070f344262c964d453c10/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L297): Missing curly brace. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00121)
1. ![CRITICAL][CRITICAL] [BridgeSupport.java#L166](https://github.com/rsksmart/rskj/blob/be1d47fc7508dac76d2070f344262c964d453c10/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L166): Missing curly brace. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00121)
1. ![CRITICAL][CRITICAL] [BridgeSupport.java#L168](https://github.com/rsksmart/rskj/blob/be1d47fc7508dac76d2070f344262c964d453c10/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L168): Missing curly brace. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00121)
1. ![CRITICAL][CRITICAL] [BridgeSupport.java#L204](https://github.com/rsksmart/rskj/blob/be1d47fc7508dac76d2070f344262c964d453c10/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L204): Missing curly brace. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00121)
1. ![CRITICAL][CRITICAL] [BridgeSupport.java#L572](https://github.com/rsksmart/rskj/blob/be1d47fc7508dac76d2070f344262c964d453c10/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L572): Missing curly brace. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00121)
1. ![CRITICAL][CRITICAL] [BridgeSupport.java#L684](https://github.com/rsksmart/rskj/blob/be1d47fc7508dac76d2070f344262c964d453c10/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L684): Missing curly brace. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00121)
1. ![CRITICAL][CRITICAL] [BridgeSupport.java#L780](https://github.com/rsksmart/rskj/blob/be1d47fc7508dac76d2070f344262c964d453c10/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L780): Missing curly brace. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00121)
1. ![CRITICAL][CRITICAL] [BridgeSupport.java#L830](https://github.com/rsksmart/rskj/blob/be1d47fc7508dac76d2070f344262c964d453c10/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L830): Missing curly brace. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00121)
1. ![CRITICAL][CRITICAL] [BridgeSupport.java#L902](https://github.com/rsksmart/rskj/blob/be1d47fc7508dac76d2070f344262c964d453c10/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L902): Missing curly brace. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00121)
1. ![CRITICAL][CRITICAL] [BridgeSupport.java#L914](https://github.com/rsksmart/rskj/blob/be1d47fc7508dac76d2070f344262c964d453c10/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L914): Missing curly brace. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00121)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**